### DISCUSSION TITLE
String Number Conversion To <Type> With C++ Templates

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/STL#3190] [Ideas] [answered]
Use template specialization for string number parsing.
In this case consumer can provide custom specialization for more complex types.

**Like This:**
_namespace std {
&emsp;template<>
&emsp;complex_type stot<complex_type>(const string &string) {
&emsp;&emsp;/\* Implementation Goes Here \*/
&emsp;&emsp;return complex_type(/\* Parameters Goes Here \*/);
&emsp;}
}_

**There Is Link To Implementation Example ->** [Code-Example-[onlinegdb.com]](https://onlinegdb.com/mHp_hWOl5)

For backwards compatibility, can be created templated instantiated functions pointers.

 **Like This:**
_const auto stoi = stot<int32_t>;_

**Comment by StephanTLavavej** [comment, marked as answer]
This repo is not the correct place to submit proposals for C++ Standardization. See https://isocpp.org/std/submit-a-proposal for more info.

**Comment by ghost** [comment]
discussion is outdated
